runtime/src/bank.rs

```diff
@@ -283,28 +298,50 @@ impl AbiExample for CachedExecutors {
 
 impl Clone for CachedExecutors {
     fn clone(&self) -> Self {
+        self.clone_with_epoch(self.current_epoch)
+    }
+}
+impl CachedExecutors {
+    fn current_epoch(&self) -> Epoch {
+        self.current_epoch
+    }
+
+    fn clone_with_epoch(&self, epoch: Epoch) -> Self {
         let mut executors = HashMap::new();
-        for (key, (count, executor)) in self.executors.iter() {
-            executors.insert(
-                *key,
-                (AtomicU64::new(count.load(Relaxed)), executor.clone()),
-            );
+        for (key, (last_epoch_count, count, executor)) in self.executors.iter() {
+            // If the epoch has changed, we store the old count and reset the counts
```

Comment:
Can you add a more descriptive description of the new cache strategy here for future readers?